### PR TITLE
docs(examples): add ibis.get_backend examples

### PR DESCRIPTION
**Pull request by IndexSeek**
## Description of changes

Adds a docstring example for [ibis.get_backend()](https://ibis-project.org/reference/connection.html#ibis.get_backend), fixes parameter block, and links to the `Table.get_backend()`.

### CODE REVIEW

File: ibis/expr/api.py [hunk 1]
```diff
@@ -1743,6 +1745,25 @@ def get_backend(expr: Expr | None = None) -> BaseBackend:
...
         The Ibis backend.
 
+    Examples
+    --------
+    >>> import ibis
+
+    Get the default backend.
+
+    >>> ibis.get_backend()  # doctest: +ELLIPSIS
+    <ibis.backends.duckdb.Backend at 0x7fffbd7e1ee0>
```
Comment (cpcloud) [inline ibis/expr/api.py:1755]:
```suggestion
    <ibis.backends.duckdb.Backend object at 0x7fffbd7e1ee0>
```

File: ibis/expr/api.py [hunk 2]
```diff
@@ -1743,6 +1745,25 @@ def get_backend(expr: Expr | None = None) -> BaseBackend:
...
+
+    Get the backend for a specific expression.
+
+    >>> polars_con = ibis.polars.connect()
+    >>> t = polars_con.create_table("t", ibis.memtable({"a": [1, 2, 3]}))
+    >>> ibis.get_backend(t)  # doctest: +ELLIPSIS
+    <ibis.backends.polars.Backend at 0x7fffab10a5d0>
```
Comment (cpcloud) [inline ibis/expr/api.py:1762]:
```suggestion
    <ibis.backends.polars.Backend object at 0x7fffab10a5d0>
```

**Review by cpcloud — CHANGES_REQUESTED** [review]
Just some small adjustments to make the repr match the actual output.

File: ibis/expr/api.py [hunk 3]
```diff
@@ -1743,6 +1745,25 @@ def get_backend(expr: Expr | None = None) -> BaseBackend:
...
         The Ibis backend.
 
+    Examples
+    --------
+    >>> import ibis
+
+    Get the default backend.
+
+    >>> ibis.get_backend()  # doctest: +ELLIPSIS
+    <ibis.backends.duckdb.Backend object at 0x7fffbd7e1ee0>
```
Comment (cpcloud) [inline ibis/expr/api.py:1755]:
```suggestion
    <ibis.backends.duckdb.Backend object at 0x...>
```

File: ibis/expr/api.py [hunk 4]
```diff
@@ -1743,6 +1745,25 @@ def get_backend(expr: Expr | None = None) -> BaseBackend:
...
+
+    Get the backend for a specific expression.
+
+    >>> polars_con = ibis.polars.connect()
+    >>> t = polars_con.create_table("t", ibis.memtable({"a": [1, 2, 3]}))
+    >>> ibis.get_backend(t)  # doctest: +ELLIPSIS
+    <ibis.backends.polars.Backend object at 0x7fffab10a5d0>
```
Comment (cpcloud) [inline ibis/expr/api.py:1762]:
```suggestion
    <ibis.backends.polars.Backend object at 0x...>
```

**Review by cpcloud — CHANGES_REQUESTED** [review]
Oops, need to actually use the ellipsis 😅

File: ibis/expr/api.py (hunk 3 above)
Comment (IndexSeek) [reply ibis/expr/api.py]: I was just taking a look at that, found an example at:

https://github.com/ibis-project/ibis/blob/161c61f597b323e5f2c6b3085b545b9132514e2d/ibis/backends/impala/__init__.py#L159-L160

**Comment by IndexSeek** [comment]
Thank you for helping me work through this one! 

Later (likely this evening), I'll tweak the `Table.get_backend` method to link back to this one so that the example follows a similar style by avoiding using `type` for consistency.

File: ibis/expr/api.py (hunk 4 above)
Comment (IndexSeek) [reply ibis/expr/api.py]: Resolved by [068822e](https://github.com/ibis-project/ibis/pull/10489/commits/068822eb138cc74754087626a8b5a1408a357461)

**Review by cpcloud — APPROVED** [review]

**Pull request merged**